### PR TITLE
Feature/tests

### PR DESCRIPTION
**Pull request by RileyManda**
We have completed the task of implementing tests with RSpec according to the [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md).

The following changes were implemented: 🌱 

### Setup 🪜 
- [x] Setup/Installed RSpec 

### Tests 🧪 
Wrote tests for the following classes and decorators:

- [x] Book:Tests
- [x] Person:Tests
- [x] Student:Tests
- [x] Teacher:Tests
- [x] Classroom:Tests
- [x] Rental:Tests
- [x] Capitalize_Decorator:Tests
- [x] Trimmer_Decorator:Tests
- [x] Decorator:Tests

Please review our implementations and suggest any changes or improvements.

Thank you 🙏🏻

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  it 'initializes with a unique ID' do
+    expect(person.instance_variable_get(:@id)).to be_a(Integer)
+  end
+  it 'can use services with parent permission' do
+    person = Person.new(15, 'Bob', parent_permission: true)
+    expect(person.can_use_services?).to be_truthy
+  end
+end
```
Comment (cynthiainga) [inline spec/person_spec.rb]: - [ ] Good job so far but you have to add some tests for your person file, there are some scenarios that were missed 🙃 
  - The constructor method should return an empty rental as well but you just put name, age, a dn parent_permission,
  - You should test the scenario of an unknown name; if the name is not provided the default name should be `Unknown`. The constructor method sets a default value for the optional name parameter if it is not provided.
  - You should test the context when adding a new rental so the `add_rental` method returns a rental and adds it to the book.
  - You should test the context when checking permissions:
    - The `can_use_services?` method returns false if under age and no parent permission
    - The `can_use_services?` method returns true if over age
  - You should test the context when validating the name, the `correct_name` method corrects the name for example if the name is `alongnameiscorrected` it should be equal to `Alongnamei`

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @RileyManda & @meryemsanem, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] All implemented tests are passing ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] All test files are placed in a different directory (`/spec`) ✔️ 
- [ ] There is one file per class ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by RileyManda** [comment]
We have completed implementing the changes requested:

The constructor method returns an empty rental 
Tested the scenario of an unknown name; if the name is not provided the default name should be Unknown.
The constructor method sets a default value for the optional name parameter if it is not provided.
Tested the context when adding a new rental so the add_rental method returns a rental and adds it to the book.
Tested the context when checking permissions
Tested can_use_services? method returns false if under age and no parent permission
Tested can_use_services? method returns true if over age
Tested the context when validating the name

**Comment by RileyManda** [comment]
We have completed implementing the changes requested:

✅ The constructor method returns an empty rental 
✅ Tested the scenario of an unknown name; if the name is not provided the default name should be Unknown.
✅ The constructor method sets a default value for the optional name parameter if it is not provided.
✅ Tested the context when adding a new rental so the add_rental method returns a rental and adds it to the book.
✅ Tested the context when checking permissions
✅ Tested can_use_services? method returns false if under age and no parent permission
✅ Tested can_use_services? method returns true if over age
✅ Tested the context when validating the name

Please review our implementations and suggest any changes or improvements.

Thank you 🙏🏻

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi team 👋, 

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ All tests are placed in separate directories 📌 
✔️ All tests are green 🟢 
✔️ Good commit history

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**